ibis/backends/postgres/__init__.py

```diff
@@ -295,19 +566,221 @@ def create_schema(
+        sql = sge.Create(
+            kind="SCHEMA", this=sg.table(name, catalog=database), exists=force
+        )
+        with self._safe_raw_sql(sql):
+            pass
 
     def drop_schema(
-        self, name: str, database: str | None = None, force: bool = False
+        self,
+        name: str,
+        database: str | None = None,
+        force: bool = False,
+        cascade: bool = False,
     ) -> None:
         if database is not None and database != self.current_database:
             raise exc.UnsupportedOperationError(
                 "Postgres does not support dropping a schema in a different database"
             )
-        name = self._quote(name)
-        if_exists = "IF EXISTS " * force
-        with self.begin() as con:
-            con.exec_driver_sql(f"DROP SCHEMA {if_exists}{name}")
+
+        sql = sge.Drop(
+            kind="SCHEMA",
+            this=sg.table(name, catalog=database),
+            exists=force,
+            cascade=cascade,
+        )
+        with self._safe_raw_sql(sql):
+            pass
+
+    def create_table(
+        self,
+        name: str,
+        obj: pd.DataFrame | pa.Table | ir.Table | None = None,
+        *,
+        schema: ibis.Schema | None = None,
+        database: str | None = None,
+        temp: bool = False,
+        overwrite: bool = False,
+    ):
+        """Create a table in Postgres.
+
+        Parameters
+        ----------
+        name
+            Name of the table to create
+        obj
+            The data with which to populate the table; optional, but at least
+            one of `obj` or `schema` must be specified
+        schema
+            The schema of the table to create; optional, but at least one of
+            `obj` or `schema` must be specified
+        database
+            The name of the database in which to create the table; if not
+            passed, the current database is used.
+        temp
+            Create a temporary table
+        overwrite
+            If `True`, replace the table if it already exists, otherwise fail
+            if the table exists
+        """
+        if obj is None and schema is None:
+            raise ValueError("Either `obj` or `schema` must be specified")
+
+        if database is not None and database != self.current_database:
+            raise com.UnsupportedOperationError(
+                "Creating tables in other databases is not supported by Postgres"
+            )
+        else:
+            database = None
+
+        properties = []
+
+        if temp:
+            properties.append(sge.TemporaryProperty())
+
+        if obj is not None:
+            if not isinstance(obj, ir.Expr):
+                table = ibis.memtable(obj)
+            else:
+                table = obj
+
+            self._run_pre_execute_hooks(table)
+
+            query = self._to_sqlglot(table)
+        else:
+            query = None
+
+        column_defs = [
```

Comment:
I would consider to introduce DDL operations in the future and offload their compilation to the compiler possibly with some code reuse and the backend classes could get less noisy.

Reply:
Yeah makes sense.

Reply:
https://github.com/ibis-project/ibis/issues/7927